hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -603,7 +652,23 @@ extends Importer:
     case id: Ident =>
       Param(FldFlags.empty, fieldOrVarSym(ParamBind, id), N) :: Nil
     case InfixApp(lhs: Ident, Keyword.`:`, rhs) =>
-      Param(FldFlags.empty, fieldOrVarSym(ParamBind, lhs), S(term(rhs))) :: Nil
+      // return S(moduleParam) if t represents a module parameter
+      // return N otherwise
```

Comment:
This part of the code is very convoluted. Instead of returning an option that you pattern match immediately and duplicating the `Param`-creation code, why not just make compute the flags alone first then construct the `Param` in only one place?